pixeltable/catalog/catalog.py

```diff
@@ -584,10 +597,10 @@ def _acquire_tbl_lock(
         if for_write and not tbl_md.is_mutable:
             return None  # nothing to lock
 
-        effective_version = tbl_md.current_version if tbl_md.is_snapshot else None
+        key = TableVersionKey(tbl_id, tbl_md.current_version if tbl_md.is_snapshot else None, None)
```

Comment:
Why None? We could have two concurrent pull().

Reply:
That shouldn't matter? I don't think the return value of `_acquire_tbl_lock` ever gets used for anything that's sensitive to the value of `anchor_tbl_id`. Unless there's something I'm not understanding here.

Reply:
The semantics become very difficult to follow if all of a sudden this TV instance is a valid stand-in for the one with the anchor set. There's also a chance of divergent metadata if you have both this instance and the instance with the anchor set both (accidentally?) get used in the same transaction and be present in _tbl_versions at the same time.

Weakening the semantics will make it easier for bugs to creep in in the future, even if this code happens to work in this context right now.

Reply:
(To be clear: I'm not claiming that this is the right long-term implementation, just that there isn't an actual bug that needs to block the launch. I'm also happy to make changes now, but we should discuss first to be sure we're on the same page about what they are.)

Reply:
Leave a todo: fix replica concurrency

Also put that everywhere you use anchor_tbl_id=None that doesn't have an assert !is_replica.